fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBDatabaseFactory.java

```diff
@@ -20,204 +20,60 @@
-    // when set to true, static options have been set on the FDB instance.
-    //made volatile because multiple FDBDatabaseFactory instances can be created technically, and thus can be racy during init
-    private static volatile boolean staticOptionsSet = false;
-
-    @Nonnull
-    private FDBLocalityProvider localityProvider = FDBLocalityUtil.instance();
-
-    /* Next few null until initFDB is called */
+    long UNLIMITED_TR_TIMEOUT_MILLIS = 0L;
 
     @Nullable
-    private Executor networkExecutor = null;
-
-    @Nonnull
-    private Executor executor = ForkJoinPool.commonPool();
-
-    @Nonnull
-    private Function<Executor, Executor> contextExecutor = Function.identity();
-
-    @Nullable
-    private FDB fdb;
-    private boolean inited;
-
-    private boolean unclosedWarning = true;
-    @Nullable
-    private String traceDirectory = null;
-    @Nullable
-    private String traceLogGroup = null;
-    @Nonnull
-    private FDBTraceFormat traceFormat = FDBTraceFormat.DEFAULT;
-    private int directoryCacheSize = DEFAULT_DIRECTORY_CACHE_SIZE;
-    private boolean trackLastSeenVersion;
-    private String datacenterId;
-
-    private int maxAttempts = 10;
-    private long maxDelayMillis = 1000;
-    private long initialDelayMillis = 10;
-    private int reverseDirectoryRowsPerTransaction = FDBReverseDirectoryCache.MAX_ROWS_PER_TRANSACTION;
-    private long reverseDirectoryMaxMillisPerTransaction = FDBReverseDirectoryCache.MAX_MILLIS_PER_TRANSACTION;
-    private long stateRefreshTimeMillis = TimeUnit.SECONDS.toMillis(FDBDatabase.DEFAULT_RESOLVER_STATE_CACHE_REFRESH_SECONDS);
-    private long transactionTimeoutMillis = DEFAULT_TR_TIMEOUT_MILLIS;
-    private boolean runLoopProfilingEnabled;
-
-    //made volatile because multiple FDBDatabaseFactory instances can be created technically, and thus can be racy during init
-    private static volatile int threadsPerClientVersion = 1; //default is 1, which is basically disabled
-
-    /**
-     * The default is a log-based predicate, which can also be used to enable tracing on a more granular level
-     * (such as by request) using {@link #setTransactionIsTracedSupplier(Supplier)}.
-     */
-    @Nonnull
-    private Supplier<Boolean> transactionIsTracedSupplier = LOGGER::isTraceEnabled;
-    private long warnAndCloseOpenContextsAfterSeconds;
-    @Nonnull
-    private Supplier<BlockingInAsyncDetection> blockingInAsyncDetectionSupplier = () -> BlockingInAsyncDetection.DISABLED;
-    @Nonnull
-    private FDBRecordStoreStateCacheFactory storeStateCacheFactory = PassThroughRecordStoreStateCacheFactory.instance();
-
-    @Nonnull
-    private Function<FDBLatencySource, Long> latencyInjector = DEFAULT_LATENCY_INJECTOR;
-
-    private final Map<String, FDBDatabase> databases = new HashMap<>();
+    Executor getNetworkExecutor();
 
+    void setNetworkExecutor(@Nonnull Executor networkExecutor);
 
     @Nonnull
-    public static FDBDatabaseFactory instance() {
```

Comment:
So there's the `interface`, the `base class` and the `impl`. If we say that we must not rename the `impl` class because it's marked STABLE, we'll need to "violate" the naming standards in this package. The entire package names things with HelloWorld and HelloWorldImpl (for things that have an interface and an impl, e.g. FDBDatabaseRunner and FDBDatabaseRunnerImpl).

I don't think I need FDBDatabase to be implemented any differently (apart from changing what it does in openFDB()) so that's an easier change (going to revert FDBDatabase -> FDBDatabaseImpl).

Will look into what's needed for the factory so that existing call-sites won't break.

Reply:
By and large, the concern isn't about naming standards, but mainly about keeping the contracts with the API's stable (where they are declared to be stable -- we have other annotations, such as `UNSTABLE` and `EXPERIMENTAL` which leaves us flexibility to play around until we are certain a particular design looks good). 

As noted above, I think this can be made to work by having `FDBDatabaseFactory` be `abstract` and basically contain what the `FDBDatabaseFactoryBase` contains right now.  Then `FDBDatabaseFactoryImpl` extends `FDBDatabaseFactory` and contains what is in `FDBDatabaseFactory` right now -- so the roles are reversed.